composer/loggers/mosaicml_logger.py

```diff
@@ -93,19 +93,21 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        after_load_metadata = {}
         # Log model data downloaded and initialized for run events
         log.debug(f'Logging model initialized time to metadata')
-        self._log_metadata({'model_initialized_time': time.time()})
+        after_load_metadata['model_initialized_time'] = time.time()
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
-                    self._log_metadata({'wandb/run_url': run_url})
                     log.debug(f'Logging WandB run URL to metadata: {run_url}')
+                    after_load_metadata['wandb/run_url'] = run_url
                 else:
                     log.debug('WandB run URL not found, not logging to metadata')
-        self._flush_metadata(force_flush=True)
+        self._log_metadata(after_load_metadata)
+        self._flush_metadata()
```

Comment:
I don't think this is required since it's called at the end of `self._log_metadata`

Reply:
Just updated, we are going to force flush at the end of after load!